crates/dbsp/src/operator/group/lag.rs

```diff
@@ -1,23 +1,28 @@
+use crate::dynamic::Erase;
+use crate::operator::dynamic::filter_map::DynFilterMap;
+use crate::operator::dynamic::group::LagCustomOrdFactories;
+use crate::operator::group::custom_ord::WithCustomOrd;
 use crate::{
     dynamic::{DowncastTrait, DynData, DynPair},
     operator::dynamic::group::LagFactories,
     typed_batch::{DynOrdIndexedZSet, IndexedZSet, TypedBatch},
     utils::Tup2,
-    DBData, RootCircuit, Stream, ZWeight,
+    CmpFunc, DBData, OrdIndexedZSet, RootCircuit, Stream, ZWeight,
 };
 
 impl<B> Stream<RootCircuit, B>
 where
     B: IndexedZSet,
     B::InnerBatch: Send,
 {
-    /// Lag operator matches each row in a group with a previous row in the
-    /// same group.
+    /// Lag operator matches each row in a group with a value at the given offset
+    /// in the same group.
     ///
     /// For each key in the input stream, it matches each associated value with
-    /// a previous value (i.e., value with a smaller index according to
-    /// ascending order of values), applies projection function `project` to
-    /// it and outputs the input value along with this projection.
+    /// another value in the same group with a smaller (`offset > 0`) or greater
```

Comment:
could comment that this is supposed to implement the SQL lag/lead aggregation functions